source/event.cpp

```diff
@@ -10,10 +10,29 @@ Stop::Stop(Time a_time): Event(a_time) {}
 
 void Stop::operator()() { Scheduler::get_instance().clear(); }
 
-Arrive::Arrive(Time a_time, ILink *a_link, Packet a_packet) : Event(a_time), m_link(a_link), m_packet(a_packet) {};
+Generate::Generate(Time a_time, std::weak_ptr<IFlow> a_flow, Size a_packet_size) : Event(a_time), m_flow(a_flow), m_packet_size(a_packet_size) {}
+
+void Generate::operator()() {
+    if (m_flow.expired()) {
+        return;
+    }
+    
+    m_flow.lock()->try_to_generate(m_packet_size);
+
+    // TODO: make delay not constant (maybe try_to_generate have to return it)
+    const Time GENERATE_DELAY = 7;  // ms
```

Comment:
`types.hpp` already contains information about time type
```suggestion
    const Time GENERATE_DELAY = 7;
```